### PR TITLE
AUC logloss revisit

### PR DESCRIPTION
**Pull request by yueguoguo**
### Description
An update on #580 that incorporates some suggestions from @anargyri 

### Related Issues
#445 

### Checklist:

- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [x] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: reco_utils/evaluation/python_evaluation.py [hunk 1]
```diff
@@ -677,19 +674,11 @@ def logloss(
         rating_true, rating_pred, col_user, col_item, col_rating, col_prediction
     )
 
-    # To make sure that there is no N/A in the logloss metric calculation.
-    rating_true_pred[PREDICTION_COL] = (
-        rating_true_pred[PREDICTION_COL].apply(lambda x: max(min(x, 1.0 - 10e-12), 10e-12))
-    )
```
Comment (anargyri) [inline reco_utils/evaluation/python_evaluation.py]: Why did you remove this completely? It is good to clip the predictions. But you can do it more simply with numpy.clip.

Comment (yueguoguo) [reply reco_utils/evaluation/python_evaluation.py]: Didn't I leave any comment that explain this somewhere?... Actually the `sklearn.metrics.log_loss` take care of the clipping.

Comment (anargyri) [reply reco_utils/evaluation/python_evaluation.py]: Great then.
